coalib/misc/Annotations.py

```diff
@@ -0,0 +1,44 @@
+
+
+def typechain(*args):
+    """
+    Returns function which applies the first transformation it can from args
+    and returns transformed value, raises ValueError when cannot apply none.
+
+    >>> function = typechain(int, ord, None)
+    >>> function("10")
+    10
+    >>> function("a")
+    97
+    >>> function(int)
+    <class 'int'>
+    >>> function(None) is None
+    True
+    >>> function("str")
+    Traceback (most recent call last):
+        ...
+    ValueError: Couldn't convert to any specified type value 'str'.
+
+    :raises TypeError:  Raises when either no functions are specified for
+                        checking or one or more object isn't callable and not
+                        None.
+    """
+    if len(args) == 0:
+        raise TypeError("No arguments were provided.")
```

Comment:
But, will it be the desired behaviour for someone to pass an empty list to `typechain` or unintended? In the first case, I'd agree that it's then OK, but still I cannot come up with situation when something like that would be intended. In the second case, it would be better to raise exception ASAP.